cmake/modules/SwiftSetIfArchBitness.cmake

```diff
@@ -6,39 +6,45 @@ function(set_if_arch_bitness var_name)
       "" # multi-value args
       ${ARGN})
 
-  if("${SIA_ARCH}" STREQUAL "avr")
+  # FIXME: This is a workaround for Concurrency/CMakeLists.txt setting the GYB_SOURCES in embedded-concurrency
```

Comment:
@etcwilde @kubamracek I'm not sure what I'm doing wrong in `Runtimes/Core/Concurrency/CMakeLists.txt` that the SIA_ARCH ends up like this when we try to add `GYB_SOURCES` in the

```
    add_swift_target_library_single(
      embedded-concurrency-${mod}
```

It seems very much the same like all other modules, like Synchronization, which do this for their gyb as well... Would you be able to help me out so we don't need to do this workaround? 

I'm not sure if this workaround is "fine" or would break something in the embedded build for example.

Reply:
This function isn't used in the new build system so this definitely isn't coming from changes in `Runtimes/Core/Concurrency/CMakeLists.txt`. It would be coming from how the gyb bits work in `add_swift_target_library`.

Reply:
But it is in the old build, so either way need it to work right?

Reply:
Hmm, so looking at add_swift_target_library there's not much there but handle_gyb_sources does a small dance with ARCH... I'll see if we need to change anything there or not maybe